src/screens/WalletVersionError.js

```diff
@@ -30,89 +23,85 @@ const mapDispatchToProps = dispatch => {
  *
  * @memberof Screens
  */
-class WalletVersionError extends React.Component {
-  static contextType = GlobalModalContext;
-  constructor(props) {
-    super(props);
+function WalletVersionError() {
+  const context = useContext(GlobalModalContext);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const alertSuccessRef = useRef();
 
-    this.alertSuccessRef = React.createRef();
+  /**
+   * Called when backup of words succeed, then close modal and show success message
+   */
+  const backupSuccess = () => {
+    context.hideModal();
+    LOCAL_STORE.markBackupDone();
+    dispatch(updateWords(null));
+    alertSuccessRef.current.show(3000);
   }
 
   /**
    * Called if user clicks the button to do the words backup
    *
    * @param {Object} e Event emitted when clicking on the button
    */
-  backupClicked = (e) => {
+  const backupClicked = (e) => {
     e.preventDefault();
 
-    this.context.showModal(MODAL_TYPES.BACKUP_WORDS, {
+    context.showModal(MODAL_TYPES.BACKUP_WORDS, {
       needPassword: true,
-      validationSuccess: this.backupSuccess,
+      validationSuccess: backupSuccess,
     });
   }
 
   /**
-   * Called when backup of words succeed, then close modal and show success message
+   * Called when reset wallet succeed, then close modal and go to welcome screen
    */
-  backupSuccess = () => {
-    this.context.hideModal();
-    LOCAL_STORE.markBackupDone();
-    this.props.updateWords(null);
-    this.alertSuccessRef.current.show(3000);
+  const handleReset = () => {
+    context.hideModal();
+    dispatch(walletReset());
```

Comment:
Note: a bug was found while testing this. The redux `wallet` object is not available on the store during the context of this screen.

So, when the relevant saga method tries to reset the wallet, it finds nothing to reset and the operation is not executed properly.

Since the scope of this PR is only to refactor this screen to a functional component, this fix will be implemented in a future moment.

Reply:
Did you create an issue describing this?